src/nvim/api/buffer.c

```diff
@@ -395,10 +395,10 @@ void nvim_buf_set_lines(uint64_t channel_id,
     mark_adjust((linenr_T)start, (linenr_T)(end - 1), MAXLNUM, extra);
   }
 
-  changed_lines((linenr_T)start, 0, (linenr_T)end, extra);
+  changed_lines((linenr_T)start, 0, (linenr_T)end, (long)extra);
```

Comment:
I would rather make it `ptrdiff_t`: as it is being set based on `size_t` values `ptrdiff_t` is more logical choice.